alibi/explainers/tests/test_ale.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from sklearn.datasets import load_boston, load_iris
+from sklearn.linear_model import LinearRegression, LogisticRegression
+import pytest
+from alibi.explainers.ale import ale_num, adaptive_grid, get_quantiles
+from alibi.api.defaults import DEFAULT_DATA_ALE, DEFAULT_META_ALE
+
+
+@pytest.mark.parametrize("min_bin_points", [1, 4, 10])
+def test_ale_num_linear_regression(min_bin_points):
+    """
+    The slope of the ALE of linear regression should equal the learnt coefficients
+    """
+    X, y = load_boston(return_X_y=True)
```

Comment:
I'm more of a fan of the approach where one has a fixture for data, one for the model an combines the two rather than hardcoding in the test. The test needs to check something about ale function, and it's nice if we can abstract the model and the data away (maybe there comes a time when we need to check why this fails in production and it would be nice to have the ability to add datasets/models). This results in harder work as additional code as other parts of the codebase show, but makes testing more flexible (but maybe harder to implement the first time).

Reply:
I agree with this, we will have to rerorganize testing utilities and fixtures at some point as there is quite a lot going on already. I will try to make this change using fixtures, it seems a bit involved as I need access both to the model and the dataset it's parametrized by.

Reply:
I've re-parametrized tests using some new and existing fixtures, please have a look. Also, I haven't updated `SUPPORTED_DATASETS` and `get_dataset` as they were missing some values already and I wasn't clear where these are used in the tests?